File import uses suggested types for preview, and table patch endpoint for updating columns

pavish opened the pull request:
Fixes #568 
Fixes #295 

This PR:
* Uses `type_suggestions` endpoint for selecting types during preview
* Uses table patch endpoint instead of columns API while saving table from preview

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. --> 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

kgodey reviewed (changes requested):
@pavish The types functionality looks good, but if I close the tab with the preview and re-open the table, it shows the table like it's already imported instead of in the preview state. Since the `import_verified` attribute is set to `false`, it should show the table in the preview state.

pavish commented:
@kgodey I've updated the PR to open preview state, when `import_verified` is `false`.

I've had to make a few changes in the backend as well:
1. Preload table list for current schema
2. Changed table list filter to accept schema id instead of schema name
   - Same schema name may be present across dbs. It is better to accept schema id, as filter.

kgodey reviewed (approved):
Looks good, thanks. I made a couple of small changes:

- Updated the name of the table list function to be `get_table_list` instead of `get_tables_list` for consistency.
- Removed the `tables` property in the schemas API.
- Updated the filter code to remove some unnecessary code (this is not related to this PR, I just noticed it while reviewing)
